pyomo/repn/tests/test_util.py

```diff
@@ -717,14 +717,14 @@ class UnknownExpression(NumericExpression):
             DeveloperError, r".*Unexpected expression node type 'UnknownExpression'"
         ):
             end[node.__class__](None, node, *node.args)
-        self.assertEqual(len(end), 8)
+        self.assertEqual(len(end), 9)
```

Comment:
What is this testing? Why did it change?

Reply:
This changed because the ExitNodeDispatcher changed a bit: by getting rid of the extra registration function (and implementing it's logic in `__missing__`, we changed how we handle unknown types: they are now registered witht he dispatcher using the `unexpected_expression_type` handler (so that handler can be overwritten).  The test now reflects that the number of dispatchers registered in the END will increase.